src/components/TreeViewFinder/TreeViewFinder.jsx

```diff
@@ -187,6 +189,45 @@ const TreeViewFinder = (props) => {
         // will proc onNodeSelect then ...
     };
 
+    useEffect(() => {
+        if (defaultSelected?.length > 0) {
+            setSelected((oldSelectedNodes) => [
+                ...oldSelectedNodes,
+                ...defaultSelected,
+            ]);
+        }
+    }, [defaultSelected]);
+
+    useEffect(() => {
+        if (defaultExpanded?.length > 0) {
+            setExpanded((oldExpandedNodes) => [
+                ...oldExpandedNodes,
+                ...defaultExpanded,
+            ]);
+        }
+    }, [defaultExpanded]);
```

Comment:
`default*` prop naming should be used to initialize component state that's all. If you start to have effects like this, it doesn't respect this naming rule (check in this MUI component for example between `defaultExpanded` and `expanded` props : https://mui.com/material-ui/api/accordion/#accordion-prop-defaultExpanded)
then it becomes a controlled component. And you should remove the local state `expanded` and be controlled by props passing by the user of this component.

Reply:
same for `defaultSelected` of course